File: backend/app/api/v1/routers/ingest.py
"""Data ingestion API router."""
from fastapi import APIRouter, UploadFile, File, HTTPException
from app.services.ingestion import ingest_csv_data

router = APIRouter()


@router.post("/csv")
async def upload_csv(file: UploadFile = File(...)):
    """Upload and ingest CSV file."""
    if not file.filename.endswith('.csv'):
        raise HTTPException(status_code=400, detail="Invalid file type. Please upload a CSV.")

    content = await file.read()
    count = await ingest_csv_data(content)
    return {"message": f"Successfully ingested {count} assets."}

File: frontend/src/api/ingest.ts
import { httpClient } from '../lib/httpClient';

export interface CSVIngestResponse {
  message: string;
}

export const ingestApi = {
  /**
   * Upload CSV file for asset ingestion
   * POST /ingest/csv
   *
   * @param file - CSV file to upload
   * @returns Response with ingestion message
   */
  uploadCSV: async (file: File): Promise<CSVIngestResponse> => {
    const formData = new FormData();
    formData.append('file', file);

    const response = await httpClient.post<CSVIngestResponse>('/ingest/csv', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });

    return response.data;
  },
};
